src/Compiler/test/Microsoft.NET.Sdk.Razor.SourceGenerators.Tests/RazorSourceGeneratorTests.cs

```diff
@@ -60,6 +60,42 @@ protected override void BuildRenderTree(global::Microsoft.AspNetCore.Components.
             Assert.Single(result.GeneratedSources);
         }
 
+        [Fact, WorkItem("https://github.com/dotnet/razor/issues/8610")]
+        public async Task SourceGenerator_RazorFiles_UsingAlias_NestedClass()
```

Comment:
Test from https://github.com/dotnet/razor/pull/8614 to verify it's also fixed here.